### PR TITLE
Creating datalinks from code

### LINKED ISSUE #9673: Creating datalinks in Enso Cloud programmatically
Currently there's no way to create a datalink from Enso code.

It is not only useful for users and library creators but it is also needed so that our datalink tests could be made self-sufficient, not expecting an existing setup on the test account.

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Closes #9673 
- Adds the ability to save an existing Postgres connection as a datalink into the Enso Cloud, automatically promoting plain-text passwords into a Secret.
- Fixes dataflow error propagation in `JS_Object.from_pairs`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_Secret.enso [hunk 1]
```diff
@@ -44,17 +44,19 @@ type Enso_Secret
...
-                body = JS_Object.from_pairs [["name", name], ["value", value]]+parent_id_pair
-                handle_already_exists _ =
-                    message = "A secret with the name " + name.pretty + " already exists inside of directory " + parent_dir.name + "."
-                    Error.throw (Illegal_Argument.Error message)
-                error_handlers = Map.from_vector [["resource_already_exists", handle_already_exists]]
-                id = Utils.http_request_as_json HTTP_Method.Post Utils.secrets_api body error_handlers=error_handlers
-                Enso_Secret.Value name id
+                path = if name.contains "/" then Error.throw (Illegal_Argument.Error "Secret name cannot contain `/`.") else parent_dir.enso_path.resolve name
+                path.if_not_error <|
+                    # TODO verify if root needs special handling, maybe file a bug
```
Comment (AdRiley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_Secret.enso]: Did we do this TODO?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Errors.enso [hunk 2]
```diff
@@ -44,10 +45,14 @@ type Enso_Cloud_Error
...
-        Enso_Cloud_Error.Invalid_Response_Payload cause -> "Internal error: A response from Enso Cloud could not be parsed: " + cause.to_display_text
-        Enso_Cloud_Error.Unauthorized -> "Enso Cloud credentials file was found, but the service responded with 401 Unauthorized. You may try logging in again and restarting the workflow."
-        Enso_Cloud_Error.Connection_Error cause -> "Error connecting to Enso Cloud: " + cause.to_display_text
+        Enso_Cloud_Error.Unexpected_Service_Error uri status_code payload ->
+            "Unexpected error response from Enso Cloud: endpoint " + uri.to_text + " replied with status " + status_code.to_text + ": " + payload
+        Enso_Cloud_Error.Invalid_Response_Payload cause ->
+            "Internal error: A response from Enso Cloud could not be parsed: " + cause.to_display_text
+        Enso_Cloud_Error.Unauthorized ->
+            "Enso Cloud credentials file was found, but the service responded with 401 Unauthorized. You may try logging in again and restarting the workflow."
```
Comment (AdRiley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Errors.enso]:
```suggestion
            "Enso Cloud credentials file was found, but the service responded with Unauthorized. You may try logging in again and restarting the workflow."
```
401 doesn't seem so friendly to an end user

**Review by AdRiley — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_Secret.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_Secret.enso]: No, thanks for bringing it up - I meant to do it, I think on Tuesday, but kept forgetting... Actually, originally I meant to do it weeks ago 🤦 

I will try to check it before merging this PR. I'll check it once I'm back in office (currently on train with weak internet and I need to download an IDE build to get re-authenticated on my travel laptop as my creds expired).

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_Secret.enso]: Thanks for the reminder - after all I have removed the TODO and the workaround as after checking it turns out it's no longer needed.

**Pull request merged**